src/VisualStudio/Core/Def/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -189,8 +189,8 @@ static VsTextSpan GetVsTextSpanFromPosition(SourceText text, int position, int v
         documentId = workspace.GetDocumentIdInCurrentContext(documentId);
 
         var solution = workspace.CurrentSolution;
-        var document = solution.GetDocument(documentId);
-        if (document == null)
+        var textDocument = await solution.GetRequiredTextDocumentAsync(documentId, cancellationToken).ConfigureAwait(false);
+        if (textDocument is SourceGeneratedDocument)
```

Comment:
Since you have the generated document from the Solution, is it still necessary to get it again from the Project?

Reply:
Oh woops, good catch. It was a null check before but no need to get from project now

Reply:
Also more generally behavior changed here: if you passed in a documentId before for a document that went missing, this function would no-op. Now it'll throw. Given the comment on line 198 around stale documents, I imagine this was a fix made for exactly this sort of scenario happening.